fix(parser): correctly fail parsing when parsing `foo.bar?.`

camc314 opened the pull request:
closes #9497

[swc](https://play.swc.rs/?version=1.11.5&code=H4sIAAAAAAAAA8tJLVEoSi0oylewVaiuteYCs%2FXS7PUAMGIdLBkAAAA%3D&config=H4sIAAAAAAAAA1WPSw7DIAwF9zkF8rrbdtE79BAWdSIifrKJVBTl7iUE0maH3xsz8jooBbNoeKq1PMsQkYX4nEsi2Sf8lARIOxTNJia49XaWvRrRCtVoOxpIyBOluiX3hoMNQajjLXPGmzH%2FC3VwkUnkCu4o%2BsnSVTc0JbjwXmrZDkk50qF%2FwA%2FqsvNjMPLqm4kXGrYvhlQioBQBAAA%3D)

[typescript](https://www.typescriptlang.org/play/?#code/PTAEAEBcGcFoDsD2BjAFgU2QawFABt1JQAndAB2MVAF5QBvAXwG4dSLEA6AMwH4Og)

[babel](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=DYUwLgBATiAOUHsIF4IG8C-BuAUDeCAdAGYD8hQA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2%2Cstage-3&prettier=false&targets=&version=7.26.9&externalPlugins=&assumptions=%7B%7D)

[esbuild](https://esbuild.github.io/try/#dAAwLjI1LjAAAGxldCByZXBybyA9IHt9OwpyZXByby5mPy4)

camc314 commented:
* **#9499** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9499?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9499?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9499).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Mar 3, 6:57 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camc314 committed:
fix(parser): correctly fail parsing when parsing `foo.bar?.` (#9499)

closes #9497

[swc](https://play.swc.rs/?version=1.11.5&code=H4sIAAAAAAAAA8tJLVEoSi0oylewVaiuteYCs%2FXS7PUAMGIdLBkAAAA%3D&config=H4sIAAAAAAAAA1WPSw7DIAwF9zkF8rrbdtE79BAWdSIifrKJVBTl7iUE0maH3xsz8jooBbNoeKq1PMsQkYX4nEsi2Sf8lARIOxTNJia49XaWvRrRCtVoOxpIyBOluiX3hoMNQajjLXPGmzH%2FC3VwkUnkCu4o%2BsnSVTc0JbjwXmrZDkk50qF%2FwA%2FqsvNjMPLqm4kXGrYvhlQioBQBAAA%3D)

[typescript](https://www.typescriptlang.org/play/?#code/PTAEAEBcGcFoDsD2BjAFgU2QawFABt1JQAndAB2MVAF5QBvAXwG4dSLEA6AMwH4Og)

[babel](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=DYUwLgBATiAOUHsIF4IG8C-BuAUDeCAdAGYD8hQA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2%2Cstage-3&prettier=false&targets=&version=7.26.9&externalPlugins=&assumptions=%7B%7D)

[esbuild](https://esbuild.github.io/try/#dAAwLjI1LjAAAGxldCByZXBybyA9IHt9OwpyZXByby5mPy4)
